dips/dip-1.mdx

```diff
@@ -211,18 +211,88 @@ Represents an error that occurred in response to a Command.
     "field": "payment.sender.kyc_data.surname",
-    "code": "missing_data",
+    "code": "missing_field",
     "message": "",
 }
 ```
-`command_error` occurs in response to a Command failing to be applied - for example, a high level validation error.  `protocol_error` occurs in response to a failure related to the lower-level protocol.  Note: Additional information detailing all errors will be added.
 
+`command_error` occurs in response to a Command failing to be applied - for example, a high level validation error.  `protocol_error` occurs in response to a failure related to the lower-level protocol.
+
+#### List of Error Codes
+
+##### HTTP Header Validation Error Codes
+
+`invalid_x_request_sender_address`:
+* `X-REQUEST-SENDER-ADDRESS` header value is not the request sender’s address in the command object. All command objects should have a field that is the request sender’s address. For payment object, it is `sender#address` or `receiver#address`.
+* Could not find Diem's onchain account by the `X-REQUEST-SENDER-ADDRESS` header value.
+* Could not find the compliance key of the onchain account found by the `X-REQUEST-SENDER-ADDRESS` header value.
+* The compliance key found from the onchain account is not a valid ED25519 public key.
+
+`missing_http_header`: missing HTTP header `X-REQUEST-ID` or `X-REQUEST-SENDER-ADDRESS`.
+
+##### JWS Validation Error Codes
+
+`invalid_jws`: invalid JWS format (compact) or protected header
+
+`invalid_jws_signature`: JWS signature verification failed
+
+##### Request Object Validation Error Codes
+
+`invalid_json`: decoded JWS body is not json.
+
+`invalid_object`: command request/response object json is not object, or the command object type does not match `command_type`.
+
+`missing_field`:
+* Missing required field.
+* An optional field is required to be set for a specific state, e.g. PaymentObject requires sender's `kyc_data` (which is an optional field for `PaymentActorObject`) when sender init the `PaymentObject`.
+
+`unknown_field`: field is unknown for an object.
+
+`unknown_command_type`: invalid/unsupported `command_type`.
+
+`invalid_field_value`:
+* Invalid / unknown enum field values.
+* UUID field value does not match UUID format.
+* Payment actor address is not a valid DIP-5 account identifier.
+* Currency field value is not a valid Diem currency code for the connected network.
+
+`invalid_command_producer`: The HTTP request sender is not the right actor to send the payment object. For example, if the actor receiver sends a new command with payment object change that should be done by actor sender.
+
+`invalid_initial_or_prior_not_found`: could not find command by reference_id for a non-initial state command object; for example, actor receiver received a payment command object that actor sender status is `ready_for_settlement`, but receiver could not find any command object by the reference id.
+
+`no_kyc_needed`: payment action amount is under travel rule limit.
```

Comment:
Will this actually be used? Whoever triggers a travel rule interaction is probably doing it because they want KYC info, whether it's above or below threshold?

Reply:
The error codes are for the validation result of request receiver, it can't assume sender has no bug.
From receiver's point view, it should avoid unnecessary kyc data exchange which may cause data leak.